smac_planner/src/a_star.cpp

```diff
@@ -83,11 +83,7 @@ void AStarAlgorithm::setCosts(
     _x_size = x;
     _y_size = y;
     int x_size_int = static_cast<int>(_x_size);
-    _von_neumann_neighborhood = {
-      -1, +1, -x_size_int, +x_size_int};
-    _moore_neighborhood = {
-      -1, +1, -x_size_int, +x_size_int, -x_size_int - 1,
-      -x_size_int + 1, +x_size_int - 1, +x_size_int + 1};
+    Node::initNeighborhoods(x_size_int, _neighborhood);
```

Comment:
I wonder how this will generalize when we add in the hybrid stuff. I'm not sure this is a good long term move, but good for now

Reply:
Or maybe we just leave this since you did it already. @carlosluisg let me know what you think here. I'm not really a fan of the static stuff but it seems like the best option right now. In line with the comment 2 more below, maybe we do the metrics and see if it had any negative impacts. If its all good, then we should keep it, but open for suggestions.

Reply:
I expect some refactoring needed once we introduce the new nodes for Hybrid A*. For example, it may be hard to find a general definition of neighbors as indices offsets in the graph, as we currently do. For now I deemed it was better to get things moving and iterate on the design once we introduce more complexity into it.

The static method was just so that we only have one neighborhood instance shared among all nodes (since in the 2D case the neighborhood doesn't change). For the hybrid node we will need a method to compute the neighborhood based on its continuous state (x, y, theta).

Reply:
Makes sense, I suppose we can keep it for now. I agree that something will likely need to be shuffled around. I'm starting to come to terms with the need to potentially separate the getNeighbors from the Node template for cleanliness and efficiency. We may choose to have the specific  `NodeT` have an enum for which neighborhoods are valid so that we can avoid an extra parameter.

I suppose my recommendation would be to revert the getNeighbors back to the A* algorithm since I think that's what we'll have to do in the long run and what we had before. The single list doesn't work with hybrid-A* since we're sampling in continuous space not just 4 left, 10 up, rotated 2 quantization, its going to be continuous coordinates that we apply to a discrete cell.